sdk/batch/batch-rest/src/batchClient.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getClient, ClientOptions } from "@azure-rest/core-client";
+import { logger } from "./logger.js";
+import { TokenCredential, AzureNamedKeyCredential, isTokenCredential } from "@azure/core-auth";
+import { BatchClient } from "./clientDefinitions.js";
+import { createBatchSharedKeyCredentialsPolicy } from "./credentials/batchSharedKeyCredentials.js";
+import { createReplacePoolPropertiesPolicy } from "./replacePoolPropertiesPolicy.js";
+
+/**
+ * Initialize a new instance of `BatchClient`
+ * @param endpointParam - Batch account endpoint (for example: https://batchaccount.eastus2.batch.azure.com).
+ * @param credentials - uniquely identify client credential
+ * @param options - the parameter for all optional parameters
+ */
+export default function createClient(
+  endpointParam: string,
+  credentials: TokenCredential | AzureNamedKeyCredential,
+  options: ClientOptions = {},
+): BatchClient {
+  const endpointUrl = options.endpoint ?? options.baseUrl ?? `${endpointParam}`;
+  options.apiVersion = options.apiVersion ?? "2024-02-01.19.0";
+  const userAgentInfo = `azsdk-js-batch-rest/1.0.0-beta.1`;
+  const userAgentPrefix =
+    options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+      ? `${options.userAgentOptions.userAgentPrefix} ${userAgentInfo}`
+      : `${userAgentInfo}`;
+  options = {
+    ...options,
+    userAgentOptions: {
+      userAgentPrefix,
+    },
+    loggingOptions: {
+      logger: options.loggingOptions?.logger ?? logger.info,
+    },
+    telemetryOptions: {
+      clientRequestIdHeaderName:
+        options.telemetryOptions?.clientRequestIdHeaderName ?? "client-request-id",
+    },
+    credentials: {
+      scopes: options.credentials?.scopes ?? ["https://batch.core.windows.net//.default"],
+    },
+    additionalPolicies: [
+      ...(options?.additionalPolicies ?? []),
+      // TODO: remove after service remove certificate feature completely
```

Comment:
what does the TODO here mean?

Reply:
Our service has retried certificate related feature, but certificate models are still in our Swagger, and there is one API required to pass in a field call `certificateReferences` which is a list of certificate. The API board recommended us to remove certificate models in TypeSpec, so the generated code won't the `certificateReferences` fields, which makes the API call failed (`certificateReferences` is required).

This additional policy here is to add `certificateReferences` fields back and assign it an empty list, this policy can be removed after service remove `certificateReferences` field,